### PR TITLE
Fix credit grant deduplication to use stable featureId

### PR DESCRIPTION
**Pull request by joeysabs**
## What Does this PR Do?

Fixes a security vulnerability in credit grant deduplication by replacing ephemeral `subscription_item_feature.id` with the stable `featureId` for deduplication checks. This prevents customers from exploiting rapid downgrade/upgrade cycles to receive duplicate credit grants within the same billing period.

### Changes
- Modified `grantProratedCreditsForFeatures` to use a JOIN query that checks by stable `featureId` instead of ephemeral subscription item feature IDs
- Updated deduplication logic to prevent credit grants when the same product feature already has credits in the current billing period
- Added required drizzle-orm imports (`and`, `eq`, `inArray`) and schema imports

### Impact
Customers can no longer receive duplicate mid-period credit grants through subscription adjustments. The first adjustment grants credits for a feature, and subsequent adjustments within the same billing period correctly identify existing credits and skip re-granting.

---
## Summary by cubic
Fixes credit grant deduplication by checking the stable featureId instead of the ephemeral subscription item feature ID. Prevents duplicate credits from rapid plan changes in the same billing period and only deduplicates against ManualAdjustment credits.

- **Bug Fixes**
  - Deduplicates via JOIN on subscriptionItemFeatures.featureId and filters for sourceReferenceType = ManualAdjustment (ignores BillingPeriodTransition credits).
  - Filters by usageMeterId when present and adds an early guard when no stable feature IDs exist to avoid SQL errors.

<sup>Written for commit a67141f889bf456afe2ddaf02ce9d8f1ccef951d. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented duplicate mid-period credit issuance by deduplicating on stable feature IDs and honoring optional usage-meter filtering; improved prorated credit accuracy across multi-feature subscriptions and repeated mid-period adjustments.
* **Tests**
  * Added comprehensive tests covering stable-feature deduplication, multi-feature interactions, repeated adjustments, proration calculations, ledger/credit records, expirations, tenant isolation, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/subscriptions/subscriptionItemHelpers.test.ts [hunk 1]
```diff
@@ -1599,5 +1599,1051 @@ describe('subscriptionItemHelpers', () => {
...
+            )
+            const finalOnceCredits = allOnceCredits.filter(
+              (c) => c.issuedAmount === 500
+            )
+            expect(finalOnceCredits.length).toBe(1)
+            expect(finalOnceCredits[0].expiresAt).toBeNull()
+          })
+        })
+
+        it('should handle mix of Once and EveryBillingPeriod features in same subscription', async () => {
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/subscriptions/subscriptionItemHelpers.test.ts]: +1

File: platform/flowglad-next/src/subscriptions/subscriptionItemHelpers.test.ts [hunk 2]
```diff
@@ -1599,5 +1599,1051 @@ describe('subscriptionItemHelpers', () => {
...
+            // Should have 2 credits (one for each feature type)
+            expect(result1.usageCredits.length).toBe(2)
+
+            const onceCredit = result1.usageCredits.find(
+              (c) => c.issuedAmount === 500
+            )
+            const everyBillingPeriodCredit =
+              result1.usageCredits.find((c) => c.issuedAmount === 50)
+
+            expect(onceCredit).toBeDefined()
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/subscriptions/subscriptionItemHelpers.test.ts]: +1

**Review by agreea — COMMENTED** [review]
This one might need another round
